cryptoservice/crypto_service_test.go

```diff
@@ -300,6 +300,40 @@ func (c CryptoServiceTester) TestListFromMultipleKeystores(t *testing.T) {
 	}
 }
 
+// asserts that adding a key adds to all keystores
+// and adding an existing key either succeeds if the role matches or fails if it does not
+func (c CryptoServiceTester) TestAddKey(t *testing.T) {
+	cryptoService := c.cryptoServiceFactory()
+	cryptoService.keyStores = append(cryptoService.keyStores,
+		trustmanager.NewKeyMemoryStore(passphraseRetriever))
+
+	privKey, err := trustmanager.GenerateECDSAKey(rand.Reader)
+	assert.NoError(t, err)
+
+	// Add the key to the targets role
+	assert.NoError(t, cryptoService.AddKey(privKey, data.CanonicalTargetsRole))
+
+	// Check that we added the key and its info to both keystores
```

Comment:
Also apologies, this isn't getting run.  :|  It needs to be added to the `testCryptoService` function below